deepinv/datasets/datagenerator.py

```diff
@@ -58,69 +95,168 @@ def __init__(
+                            is not None
+                        ):
+                            stacking_index = int(stacking_needle)
+                            # If the stacking index is in the target range
+                            if stacking_index in range(stacked):
+                                if marked_stacked:
+                                    registered_as_measurements = True
+                                    y[stacking_index] = member
+                                else:
+                                    warn(
+                                        f"Dataset not marked as stacked but found stacked member {member_name}. There is probably an error with the dataset.",
+                                        UserWarning,
+                                        stacklevel=2,
+                                    )
+                            else:
+                                warn(
+                                    f"Found stacked measurement member {member_name} with stacking index {stacking_index} outside of the expected range [0, {stacked}). There is probably an error with the dataset.",
+                                    UserWarning,
+                                    stacklevel=2,
+                                )
+
+                    # Register the forward operator parameters
+                    if params is not None and not registered_as_measurements:
+                        params[attr_name] = member
+
+        # Process ground truths
+        if x is not None:
+            self.x = x
+
+        # Process measurements
+        if marked_stacked:
+            if None in y:
+                raise ValueError(
+                    f"Dataset marked as stacked with {stacked} parts but some parts are missing."
+                )
+        self.y = y
 
-        if self.load_physics_generator_params:
-            self.params = {}
-            for k in self.hd5:
-                if suffix in k and k not in (f"x{suffix}", f"y{suffix}"):
-                    self.params[k.replace(suffix, "")] = self.hd5[k]
+        # Process forward operator parameters
+        if params is not None:
+            self.params = params
 
-    def __getitem__(self, index):
-        r"""
-        Returns the measurement and signal pair ``(x, y)`` at the given index.
+    def __getitem__(self, index: int) -> tuple:
+        r"""Get an entry in the dataset.
 
-        If there is no training ground truth (i.e. ``x_train``) in the dataset file,
-        the dataset returns the measurement again as the signal.
+        Return the measuremet and signal pair ``(x, y)`` at the given index, in
+        the selected split. If forward operator parameters are available, it
+        returns ``(x, y, params)`` where ``params`` is a dict of parameters.
+
+        If the split contains no ground truth, it returns a scalar NaN tensor
+        as the ground truth, similarly to :class:`deepinv.training.Trainer`.
```

Comment:
instead of referencing trainer which is just something that uses datasets, I would ref the datasets user guide https://deepinv.github.io/deepinv/user_guide/training/datasets.html